Xamarin.Forms.Platform.Android/AppCompat/ImageButtonRenderer.cs

```diff
@@ -182,23 +191,49 @@ void IVisualElementRenderer.SetElement(VisualElement element)
 			ElementChanged?.Invoke(this, new VisualElementChangedEventArgs(oldElement, ImageButton));
 			ImageButton?.SendViewInitialized(Control);
 		}
-
+		
 		public override void Draw(Canvas canvas)
 		{
-			base.Draw(canvas);
-			var background = Background;
+			if (ImageButton == null)
+				return;
+
+			var backgroundDrawable = _backgroundTracker?.BackgroundDrawable;
+
+			RectF _drawableBounds = null;
+			RectF _outlineBounds = null;
 
-			if (background is RippleDrawable rd)
+			if (backgroundDrawable != null)
 			{
-				if (rd.GetDrawable(0) is BorderDrawable bd)
+				_outlineBounds = backgroundDrawable.GetPaddingBounds(canvas.Width, canvas.Height);
+				var width = (float)MeasuredWidth;
+				var height = (float)MeasuredHeight;
+
+				var widthRatio = 1f;
+				var heightRatio = 1f;
+
+				switch (ImageButton.Aspect)
 				{
-					bd.DrawOutline(canvas, canvas.Width, canvas.Height);
+					case Aspect.Fill:
+						break;
+					case Aspect.AspectFill:
+					// AspectFill isn't fully supported right now with shadows because we would need to crop the image ourselves 
```

Comment:
A warning is good I'll add that

> Could you use this clipping trick to implement this
I don't think so because 4218 is clipping the whole control to the container but in this case we're building a drawable that's already all clipped to the ImageView. So we would need to clip the drawable itself to the border when it's drawn . Drawable doesn't have a clip to bounds property so we'd need to most likely use a mask on top of the image or redraw it clipped